src/ipc/Queue.h

```diff
@@ -410,6 +425,92 @@ OneToOneUniQueue::push(const Value &value, QueueReader *const reader)
+    // in count, we can reliably report all [theIn-count, theIn) items that were
+    // queued at theSize capturing time. We may report items already pop()ed by
+    // the other side, but that is OK because pop() does not modify items -- it
+    // only increments theOut.
+    const auto count = theSize.load();
+    statOpen(os, "pushIndex", "other", count);
+    statSamples<Value>(os, theIn - count, count); // unsigned offset underflow OK
+    statClose(os);
+}
+
+/// report a sample of [start, start + size) items
+template <class Value>
+void
+OneToOneUniQueue::statSamples(std::ostream &os, const unsigned int start, const uint32_t count) const
+{
+    if (!count) {
+        os << " ";
+        return;
+    }
+
+    os << ", items: [\n";
+    // report a few leading and trailing items, without repetitions
+    const auto sampleSize = std::min(3U, count); // leading (and max) sample
+    statRange<Value>(os, start, 0, sampleSize);
+    if (sampleSize < count) { // the first sample did not show some items
+        // The `start` offset aside, the first sample reported all items
+        // below the sampleSize offset. The second sample needs to report
+        // the last sampleSize items (i.e. starting at count-sampleSize
+        // offset) except those already reported by the first sample.
+        const auto secondSampleOffset = std::max(sampleSize, count - sampleSize);
+        const auto secondSampleSize = std::min(sampleSize, count - sampleSize);
+
+        // but first we print a sample separator, unless there are no items
+        // between the samples or the separator hides the only unsampled item
+        const auto bothSamples = sampleSize + secondSampleSize;
+        if (bothSamples + 1U == count)
+            statRange<Value>(os, start, sampleSize, 1);
+        else if (count > bothSamples)
+            os << "    # ... " << (count - bothSamples) << " items not shown ...\n";
+
+        statRange<Value>(os, start, secondSampleOffset, secondSampleSize);
+    }
+    os << "  ]";
+}
+
+/// statSamples() helper that reports n items starting from (start+skippedCount)
+template <class Value>
+void
+OneToOneUniQueue::statRange(std::ostream &os, const unsigned int start, const uint32_t skippedCount, const uint32_t n) const
+{
+    assert(sizeof(Value) <= theMaxItemSize);
+    auto offset = start + skippedCount;
+    for (uint32_t i = 0; i < n; ++i) {
+        // XXX: Throughout this C++ header, these overflow wrapping tricks work
+        // only because theCapacity currently happens to be a power of 2 (e.g.,
+        // the highest offset (0xF...FFF) % 3 is 0 and so is the next offset).
+        const auto pos = (offset++ % theCapacity) * theMaxItemSize;
+        Value value;
+        memcpy(&value, theBuffer + pos, sizeof(value));
+        os << "    { ";
+        value.stat(os);
+        os << " }, # [" << skippedCount + i << "]\n"; // item's queue position
```

Comment:
Let's drop this YAML comment to avoid wasting time on review. The value is helpful for (some) humans, but it can be computed from the already posted data if it is desperately needed. IIRC, you should be able to simplify the method profile if you remove this YAML comment.

> reserved [] characters

FWIW, YAML [allows](https://yaml.org/spec/1.2/spec.html#id2780069) those characters inside YAML comments.

Reply:
Done (d07fe48).